### PR TITLE
Document the CLI environment variables used for testing

### PR DESCRIPTION
**Pull request by marckhouzam**
### What this PR does / why we need it

Document the CLI environment variables used for testing
Environment variables that are user-facing are documented in #714 

You can read the new doc in markdown at https://github.com/marckhouzam/tanzu-cli/blob/feat/docTestEnvVars/docs/dev/README.md#environment-variables-for-testing-the-cli

### Which issue(s) this PR fixes

Fixes # N/A

### Describe testing done for PR

Visual check at https://github.com/marckhouzam/tanzu-cli/blob/feat/docTestEnvVars/docs/dev/README.md#environment-variables-for-testing-the-cli
Let CI run

### Release note

```release-note
Document the CLI environment variables used for testing
```

### CODE REVIEW

File: docs/dev/README.md [hunk 1]
```diff
@@ -55,6 +55,42 @@ To run e2e tests for the repository:
 make e2e-cli-core
 ```
 
+### Environment variables for testing the CLI
+
+Some test options affecting the CLI are only available through the use of environment
+variables.  Below is the list of such test variables:
```
Comment (anujc25) [inline docs/dev/README.md:61]: Should we add additional line stating that these variables are only for testing and can be updated or removed without any prior notice (or something like that)?

**Review by anujc25 — APPROVED** [review]
LGTM.

**Comment by vuil** [comment]
lgtm, this section (as well as the one updated in #714) seems an appropriate place to reference the doc about using `tanzu config set env` as a means to persist the env vars. 
I think that detail was explained in a row in https://github.com/marckhouzam/tanzu-cli/blob/feat/docTestEnvVars/docs/full/README.md#cli-configuration.

**Review by vuil — APPROVED** [review]

File: docs/dev/README.md (hunk 1 above)
Comment (marckhouzam) [reply docs/dev/README.md:61]: How about "These variables are only meant to be used during testing and are not subject to the CLI's backwards-compatibility policy, i.e., the could be changed or removed at any time."?

**Comment by marckhouzam** [comment]
> lgtm, this section (as well as the one updated in #714) seems an appropriate place to reference the doc about using `tanzu config set env` as a means to persist the env vars. I think that detail was explained in a row in https://github.com/marckhouzam/tanzu-cli/blob/feat/docTestEnvVars/docs/full/README.md#cli-configuration.

Good idea.
How about: 
"These variables can be set in the shell or using `tanzu config set env.<VAR> <VALUE>`
(see [the documentation about this command](../full/README.md#cli-configuration))."

**Comment by marckhouzam** [comment]
I've updated the PR with both @anujc25's and @vuil's suggestions

**Pull request merged**